test/NoDelegateCall.t.sol

```diff
@@ -4,44 +4,43 @@ pragma solidity ^0.8.20;
 import {GasSnapshot} from "forge-gas-snapshot/GasSnapshot.sol";
 import {Test} from "forge-std/Test.sol";
 import {NoDelegateCallTest} from "../src/test/NoDelegateCallTest.sol";
+import {NoDelegateCall} from "../src/NoDelegateCall.sol";
 
 contract TestDelegateCall is Test, GasSnapshot {
-    error DelegateCallNotAllowed();
-
     NoDelegateCallTest noDelegateCallTest;
 
     function setUp() public {
         noDelegateCallTest = new NoDelegateCallTest();
     }
 
-    function testGasOverhead() public {
+    function test_gas_noDelegateCall() public {
         snap(
-            "NoDelegateCallOverhead",
+            "NoDelegateCall",
             noDelegateCallTest.getGasCostOfCannotBeDelegateCalled()
                 - noDelegateCallTest.getGasCostOfCanBeDelegateCalled()
         );
     }
 
-    function testDelegateCallNoModifier() public {
+    function test_delegateCallNoModifier() public {
         (bool success,) =
             address(noDelegateCallTest).delegatecall(abi.encode(noDelegateCallTest.canBeDelegateCalled.selector));
         assertTrue(success);
     }
 
-    function testDelegateCallWithModifier() public {
-        vm.expectRevert(DelegateCallNotAllowed.selector);
+    function test_delegateCallWithModifier_revertsWithDelegateCallNotAllowed() public {
+        vm.expectRevert(NoDelegateCall.DelegateCallNotAllowed.selector);
         (bool success,) =
             address(noDelegateCallTest).delegatecall(abi.encode(noDelegateCallTest.cannotBeDelegateCalled.selector));
         // note vm.expectRevert inverts success, so a true result here means it reverted
         assertTrue(success);
     }
 
-    function testCanCallIntoPrivateMethodWithModifier() public view {
+    function test_callPrivateMethodWithModifer_succeeds() public view {
         noDelegateCallTest.callsIntoNoDelegateCallFunction();
     }
 
-    function testCannotDelegateCallPrivateMethodWithModifier() public {
-        vm.expectRevert(DelegateCallNotAllowed.selector);
+    function test_delegateCallPrivateMethodWithModifier_revertsWithDelegateCallNotAllowed() public {
```

Comment:
i feel like the naming of this test and the test above are confusing. If a method is private, by definition you cannot delegate call it because its not visible externally... so what are these tests doing?

Reply:
Tried a new name. Its a bit verbose so lmk if you have a suggestion but it's testing that a private function with the delegatecall modifier will succeed (without a delegatecall) or revert (when there is a delegatecall). There is an external entrypoint that then calls the private function.

Reply:
ohhh i see 😅 thats a tough test to name lol